### PR TITLE
Build outputs are not up-to-date inputs

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.VsixBuild.targets [hunk 1]
```diff
@@ -287,7 +287,18 @@
...
+      <_InputVSIXSourceItem Remove="@(AddModules)" />
+      <_InputVSIXSourceItem Remove="$(IntermediateOutputPath)$(_SGenDllName)" />
+      <_InputVSIXSourceItem Remove="@(_DebugSymbolsIntermediatePath)" />
+      <_InputVSIXSourceItem Remove="@(DocFileItem)" />
+      <_InputVSIXSourceItem Remove="@(SatelliteDllsProjectOutputGroupOutput->'%(FinalOutputPath)')" />
+      <_InputVSIXSourceItem Remove="@(SatelliteDllsProjectOutputGroupOutput)" />
+      <_InputVSIXSourceItem Remove="$(IntermediateOutputPath)$(TargetName).pkgdef" />
+      <_InputVSIXSourceItem Remove="@(_GeneratedExtensionJson)" />
+
+      <UpToDateCheckInput Include="@(_InputVSIXSourceItem)" />
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.VsixBuild.targets:301]: Outputs of the current project build which are inputs to the VSIX are no longer treated as build inputs.

[the remark this answers is not in the document]
Comment (mmitche) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.VsixBuild.targets:301]: Does this break incremental build when building VSIXs? What's the purpose of this change?

[the remark this answers is not in the document]
Comment (sharwell) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.VsixBuild.targets:301]: This only affects whether the fast up-to-date check in Visual Studio thinks the project is up-to-date. Prior to our recent work on this target, there were significant cases where the project was not up-to-date but the build though it was. Then our original change went too far in the other direction, and now it thinks the build is out-of-date even right after a build.

[the remark this answers is not in the document]
Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.VsixBuild.targets:301]: As long as it works, I'm fine with a workaround like this, but definitely need to get this fixed in VS SDK.

**Review by mmitche — APPROVED** [review]

**Pull request merged**